ci/bitrise-complete-guide-cloud-signing.md

```diff
@@ -0,0 +1,183 @@
+
+<p align="center">
+  <img src="../images/ci-cloudSigning/2_5.png"/>
+</p> 
+
+## Environment vars setup
+In order to make everything work on Bitrise you'll need to copy the template content from [bitrise.yml](https://github.com/nodes-projects/ci-resources-ios/blob/master/bitrise_cloud_signing.yml) into the Bitrise workflow script.
+
+<p align="center">
+  <img src="../images/ci-cloudSigning/3_1.png"/>
+</p> 
+
+Go to **Workflow > Env** Vars tab. 
+You should see all your environment variables here.
+- SLACK_CHANNEL - name of the Slack channel to send a message when the build succeeds
+- RELEASE_BUILD_CONFIGURATION and TEST_BUILD_CONFIGURATION - change only if you have different names in your project.
+- BITRISE_PROJECT_PATH and BITRISE_SCHEME - change them from placeholders to your project's values. 
+
+Leave the BUILD_CONFIGURATION empty. This is assigned in the preparation steps during building.
+
+<p align="center">
+  <img src="../images/ci-cloudSigning/3_2.png"/>
+</p> 
+
+## Secrets setup
+Go to the **Workflow > Secrets** tab. Add these secrets:
+- BUILD_TRIGGER_API_TOKEN - Can be found in the **Code** tab of your app on Bitrise. This is needed to trigger the build workflows.
+- SLACK_TOKEN - Used to send notifications after successful builds. The token is universal for the Monstarlab Slack Workspace (I think). Can be found [here](https://github.com/nodes-projects/ci-resources-ios/blob/master/slackToken) or copied from another app on Bitrise. 
+
+<p align="center">
+  <img src="../images/ci-cloudSigning/4_1.png"/>
+</p> 
+
+<p align="center">
+  <img src="../images/ci-cloudSigning/4_2.png"/>
+</p> 
+
+## Code signing
+You only need to upload the development signing certificate to Bitrise. This is used to code sign an initial archive before the code signing files are downloaded from Apple and the app is resigned with them. Even this development certificate could be omitted theoretically because the cloud code signing can create a new one if it does not find it locally. This, however, creates multiple new development certificates with every new build so it is better to upload it to Bitrise.
+
+1) Download the certificate from the Apple Developer portal.
+<p align="center">
+  <img src="../images/ci-cloudSigning/5_1.png"/>
+</p>
+
+2) The certificate is in a .cer format. You will need the private key .p12 file though that is embedded inside the .cer file.
+3) Double click the certificate and install it into your Keychain.
+4) Find the cert in your Keychain. Expand it and right tap on the private key. Select ***Export <Certificate name>***. If you do not see the private key, you downloaded a certificate without it. You can either find the person who created the certificate and get the original certificate from them or you can create a new certificate. The full certificate with the private key can be downloaded from the Apple Developer portal only once.
```

Comment:
> The full certificate with the private key can be downloaded from the Apple Developer portal only once.
Are you sure about this? I'm quite sure I was able to download the same certificate multiple times from the Apple Developer portal

Reply:
You are right. I am wrong. I thought the private key is downloaded with the .cer file only for the first time. Just tried it now and looks like the private key is put into the keychain when you create the Signing Certificate and when you download the .cer file, the private key is associated with it in the keychain. I will rewrite this part a bit.